### PR TITLE
gpui: Add `truncate` and `text_ellipsis` to TextStyle

### PR DESCRIPTION
**Pull request by huacnlee**
Release Notes:

- N/A

Ref issue #4996

## Demo

```
cargo run -p gpui --example text_wrapper 
```

https://github.com/user-attachments/assets/a7fcebf7-f287-4517-960d-76b12722a2d7

### CODE REVIEW

File: crates/gpui/src/elements/text.rs [hunk 1]
```diff
@@ -244,6 +244,8 @@ struct TextLayoutInner {
     bounds: Option<Bounds<Pixels>>,
 }
 
+static ELLIPSIS: &str = "…";
```
Comment (maxdeviant) [inline crates/gpui/src/elements/text.rs:247]: Does this need to be a `static`, or can we make it a `const`?

```suggestion
const ELLIPSIS: &str = "…";
```

**Comment by maxdeviant** [comment]
This looks good, @huacnlee!

I'd like for @as-cii to take a look over once he's back.

**Review by nathansobo — APPROVED** [review]
Seems reasonable to me if it works!

**Review by maxdeviant — APPROVED** [review]

File: crates/gpui/src/text_system/line_wrapper.rs [hunk 2]
```diff
@@ -98,6 +98,32 @@ impl LineWrapper {
...
+        }
+
+        let mut char_indices = line.char_indices();
+        for (ix, c) in char_indices {
+            let char_width = self.width_for_char(c);
+            width += char_width;
+            if width > truncate_width {
+                return SharedString::from(format!("{}{}", &line[..ix], ellipsis.unwrap_or("")));
+            }
+        }
```
Comment (JunkuiZhang) [inline crates/gpui/src/text_system/line_wrapper.rs:122]: This is wrong.

**Pull request merged**